lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1357,6 +1357,40 @@ public function remove_attribute( $name ) {
 		);
 	}
 
+	/**
+	 * Checks whether the currently matched tag has a specified CSS class.
+	 *
+	 * Example:
+	 * ```php
+	 *     $tags = new WP_HTML_Tag_Processor( '<a class="button bold"></a>' );
+	 *     $tags->next_tag();
+	 *
+	 *     $tags->has_class( 'bold' );
+	 *     // true
+	 *
+	 *     $tags->has_class( 'not-bold' );
+	 *     // false
+	 * ```
+	 *
+	 * @since 6.2.0
+	 *
+	 * @param string $sought_class_name The class name to test for.
+	 */
+	public function has_class( $sought_class_name ) {
+		if ( $this->is_closing_tag ) {
+			return false;
+		}
+
+		$classes = $this->get_attribute( 'class' );
+		if ( null === $classes ) {
```

Comment:
note that if `$this->is_closing_tag()` then `get_attribute()` will return `null` as well, so we may not need to double-check `$this->is_closing_tag`